pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -753,7 +767,29 @@ def has_content(self) -> bool:
     __repr__ = _utils.dataclasses_no_defaults_repr
 
 
-ModelResponsePart = Annotated[Union[TextPart, ToolCallPart, ThinkingPart], pydantic.Discriminator('part_kind')]
+@dataclass(repr=False)
+class ToolCallPart(BaseToolCallPart):
+    """A tool call from a model."""
+
+    part_kind: Literal['tool-call'] = 'tool-call'
+    """Part type identifier, this is available on all parts as a discriminator."""
+
+
+@dataclass(repr=False)
+class ServerToolCallPart(BaseToolCallPart):
+    """A tool call from a server tool."""
+
+    model_name: str | None = None
```

Comment:
Since we're setting this to `anthropic` or `groq`, I think it should be `provider_name`